### PR TITLE
add CI/CD for python code

### LINKED ISSUE #3: feat: CI/CD
1. CI/CD that build a scala project, runs tests and scala linters
2. SparkConnect CI that builds a single fat JAR with Deequ, Tsumugi-Server and the corresponding protobuf runtime dependency
3. Simple CI that just build a Tsumugi JAR without dependencies

All the CIs except testing should contain a spark and deequ version in the name of the output JAR.

Target Scala version is 2.12 like in Deequ.

### PR DESCRIPTION
**Pull request by dinjazelena**
Closes #3

Few notes:

1. In pyproject.toml declaring versions of dependencies by `*` is kinda not obvious and can confuse. If we want to introduce any possible latest version its better just to add dependency as `poetry add ruff` and then in toml just to add `>=5.0.0` for example. Then it says which version was installed and by specifying `>=` we would always install latest.

2. Do you any matrix for os or python version?

@SemyonSinchenko

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+on: push
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5 
+      - uses: snok/install-poetry@v1
```
Comment (SemyonSinchenko) [inline .github/workflows/ci.yml:11]: I used another one..
https://github.com/SemyonSinchenko/tsumugi-spark/blob/main/.github%2Fworkflows%2Fdocs.yml#L21

Which one is better? I would like to have one action for installing poetry for all the parts of CU

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: CI-PYTHON
```
Comment (SemyonSinchenko) [inline .github/workflows/ci.yml:1]:
```suggestion
name: python-client
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dinjazelena) [reply .github/workflows/ci.yml:11]: https://github.com/abatilo/actions-poetry
https://github.com/snok/install-poetry

Honestly both looks pretty fine, maybe the one that i used is updated more frequently but i guess both are good. Which one to use?

Comment (SemyonSinchenko) [reply .github/workflows/ci.yml:11]: Let's use your one, it looks better for me. More stars, more active development, etc. Can you update the second action in the same way?

**Review by SemyonSinchenko — APPROVED** [review]
LGTM, thank you for the contribution @dinjazelena !

**Pull request merged**